miden-crypto/src/dsa/rpo_falcon512/keys/public_key.rs

```diff
@@ -1,85 +1,59 @@
 //! Public key types for the RPO Falcon 512 digital signature scheme used in Miden VM.
-//!
-//! This module defines two main types:
-//! - `PublicKey`: A commitment to a polynomial, represented as a hash of the polynomial’s
-//!   coefficients.
-//! - `PubKeyPoly`: A public key represented directly as a polynomial over FalconFelt coefficients.
-//!
-//! The `PublicKey` is used for signature verification.
-//! The `PubKeyPoly` provides the raw polynomial form of a public key.
-
-use alloc::string::ToString;
+
+use alloc::{string::ToString, vec::Vec};
 use core::ops::Deref;
 
 use num::Zero;
 
 use super::{
-    super::{LOG_N, N, PK_LEN, Rpo256},
+    super::{LOG_N, N, PK_LEN},
     ByteReader, ByteWriter, Deserializable, DeserializationError, FalconFelt, Felt, Polynomial,
     Serializable, Signature,
 };
-use crate::{Word, dsa::rpo_falcon512::FALCON_ENCODING_BITS};
+use crate::{SequentialCommit, Word, dsa::rpo_falcon512::FALCON_ENCODING_BITS};
 
 // PUBLIC KEY
 // ================================================================================================
 
-/// A public key for verifying signatures.
-///
-/// The public key is a [Word] (i.e., 4 field elements) that is the hash of the coefficients of
-/// the polynomial representing the raw bytes of the expanded public key. The hash is computed
-/// using Rpo256.
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
-pub struct PublicKey(Word);
+/// Public key represented as a polynomial with coefficients over the Falcon prime field.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct PublicKey(Polynomial<FalconFelt>);
 
 impl PublicKey {
```

Comment:
Should we add the ability to "recover" the public key from signature (similar to what we did for ECDSA)? It could look something like:
```rust
impl PublicKey {
    pub fn recover_from(message: Word, signature: &Signature) -> Self {
        ...
    }
}
```
This would be exactly the same interface as for ECDSA - though `message` would be kind of useless here.

Or would we need the message if we did the "proper" recoverable Falcon signature?

Reply:
> This would be exactly the same interface as for ECDSA - though `message` would be kind of useless here.

Indeed, but we get uniformity across DSAs, so maybe we should do it.

> Or would we need the message if we did the "proper" recoverable Falcon signature?

Yes, in key-recovery mode we would need the message as well, so maybe even more the reason to include it in the signature

Reply:
Done!
Created an issue here #472